### DISCUSSION TITLE
Add-PnPFile - Modified/ Editor field cannot be set to a specific person

### DISCUSSION THREAD

**Discussion by Max-Enrico** [pnp/powershell#671] [Q&A]
Add-PnPFile -Path $File.FullName -NewFileName $File.Targetname -Folder $Documentlist -Values @{ Editor=$SPOFileOwnerID ; Author=$SPOFileOwnerID}

Author field will be filled correctly
Editor field is even the connect user.

Tested with PnP.PowerShell 1.5.0 powershell 5.1 as well as visual studio code host 2021.2.2.

With the last version of SharePointPnPPowerShellOnline it works.

Is this a known behavior?

thank you,
Enrico

**Comment by ToddKlindt** [comment]
I've had problems like this in the past. I have had to use Set-PnPListItem to fix it after I added the file or list item. Check out example #6 for Set-PnPListItem. You have to do a system -UpdateType. I've also had to set both the Author and Editor to get either of them to be set.

**Reply by Max-Enrico** [reply]
Hi Todd, thank you. Yes, i found also the alternate with set and systemupdate. Unfortunatly i have 10k+ files per upload sequence and with a double touch of all files it is more time-consuming.

For that i jumped back to the outdated pnponline module but here i have an >250MB upload issue.
So, 
plan a) is to got a fix 
plan b) handle it with Set-PNP in a second loop ( in the return value from add-pnpfile i can't find the item id for the new created file, so i can't append the set to the add without asking by find or query the files)

**Reply by AndersRask** [reply]
I guess the reason is, that you do not have SystemUpdate on Add-PnPFile which probably would solve it...

You should be able to use UniqueId on the returned File object to retrieve list item using Get-PnPListItem and then use that ID to set it (Set-PnPListItem does not have UniqueID)

**Comment by veronicageek** [comment]
For future ref, issue #673 was created.
